apps/api/src/app/events/usecases/add-job/add-digest-job.usecase.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@nestjs/common';
+import { JobEntity, JobRepository, JobStatusEnum } from '@novu/dal';
+import { StepTypeEnum } from '@novu/shared';
+import { WorkflowQueueService } from '../../services/workflow.queue.service';
+import { AddJobCommand } from './add-job.command';
+
+@Injectable()
+export class AddDigestJob {
+  constructor(private jobRepository: JobRepository, private workflowQueueService: WorkflowQueueService) {}
+
+  public async execute(command: AddJobCommand): Promise<boolean> {
+    const data = await this.jobRepository.findById(command.jobId);
+
+    if (!data) {
+      return false;
+    }
+
+    const isValidDigestStep = data.type === StepTypeEnum.DIGEST && data.digest.amount && data.digest.unit;
+    if (!isValidDigestStep) {
+      return false;
+    }
```

Comment:
```suggestion
    const isValidDigestStep = data?.type === StepTypeEnum.DIGEST && data?.digest.amount && data?.digest.unit;
    if (!data || !isValidDigestStep) {
      return false;
    }
```
Another grouping we can do in this other internal validation.
We could even remove the need of checking if `data` exists as it would be included in `isValidDigestStep`.